src/wallet/rpcwallet.cpp

```diff
@@ -2620,6 +2634,25 @@ static UniValue createwallet(const JSONRPCRequest& request)
 
     wallet->postInitProcess();
 
+    // Encrypt the wallet if there's a passphrase
+    if (!passphrase.empty() && !(flags & WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
+        if (!wallet->EncryptWallet(passphrase)) {
+            throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: Wallet created but failed to encrypt.");
+        }
+
+        if (!create_blank) {
+            // Unlock the wallet
+            if (!wallet->Unlock(passphrase)) {
```

Comment:
Should the wallet be relocked after setting the seed?

Reply:
I've changed it to relock since that's what encryptwallet does after encryption.